visivo/models/base/context_string.py

```diff
@@ -0,0 +1,70 @@
+from typing import Any
+
+import re
+
+from visivo.models.base.parent_model import ParentModel
+
+INLINE_REF_REGEX = r"\${\s*ref\(([a-zA-Z0-9\s'\"\-_]+?)\)[\.\d\w]*\s*}"
+INLINE_REF_PROPS_PATH_REGEX = r"\${\s*ref\([a-zA-Z0-9\s'\"\-_]+?\)([\.\d\w\]\[]*)\s*}"
+INLINE_PATH_REGEX = r"\${\s*([a-zA-Z0-9\s'\"\-_\.\[\]]+?)\s*}"
+
+
+class ContextString:
+    """
+    Represents a string that contains a reference to named object in the project model.
+    Currently you can reference another object by using the ref() function.  Example:
```

Comment:
Could be good to use [admonitions](https://squidfunk.github.io/mkdocs-material/reference/admonitions/) to call out the deprecation of the old `ref()` function
ex- 
``` 
!!! warning "<v1.037 behavior" 
    In previous versions of visivo you were able to reference other models using the naked `ref()` 
    function
```

Reply:
I like this.